app/services/course/assessment/answer/text_response_auto_grading_service.rb

```diff
@@ -0,0 +1,73 @@
+class Course::Assessment::Answer::TextResponseAutoGradingService < \
+  Course::Assessment::Answer::AutoGradingService
+  def grade(auto_grading)
+    auto_grading.answer.grade, messages = grade_answer(auto_grading.answer.actable)
+    auto_grading.result = { messages: messages }
+    super(auto_grading)
+  end
+
+  private
+
+  # Grades the given answer.
+  #
+  # @param [Course::Assessment::Answer::TextResponse] answer The answer specified by the
+  #   student.
+  # @return [Array<(Fixnum, Object)>] The grade and the messages to be assigned to the grading.
+  def grade_answer(answer)
+    question = answer.question.actable
+    answer_text = answer.answer_text.downcase
```

Comment:
I think the downcase here should belong in the matcher, not at this level of abstraction.